api/proto/config/storage.proto

```diff
@@ -193,3 +193,38 @@ message Volume {
   string displayName = 7;  // Optional friendly name echo'ed in info message
   bool clear_text = 8;  // Flag to indicate the volume encryption needed or not
 }
+
+// DiskConfigType is the desired configuration of disks
+enum DiskConfigType {
+  DISK_CONFIG_TYPE_UNSPECIFIED = 0; // no configured, do nothing
+  DISK_CONFIG_TYPE_EVEOS = 1; // the disk EVE is installed on
+  DISK_CONFIG_TYPE_PERSIST = 2; // the disk is separate persist partition, not zfs
+  DISK_CONFIG_TYPE_ZFS_ONLINE = 3; // included in zfs and online
+  DISK_CONFIG_TYPE_ZFS_OFFLINE = 4; // included in zfs and offline
+  DISK_CONFIG_TYPE_APPDIRECT = 5; // for direct assignment
+  DISK_CONFIG_TYPE_UNUSED = 6; // removed from zfs/app-direct
+}
+
+//DiskDescription describes disk
+message DiskDescription {
+  string disk_name = 1;
+  string disk_serial = 2;
+}
+
+//DiskConfig describe desired configuration of disk
+message DiskConfig {
+  DiskDescription disk = 1;
+  DiskDescription old_disk = 2; // define old disk here to replace it with new one defined in disk
+  DiskConfigType disk_config = 3;
+}
+
+//DisksConfig is a configuration of disks
+//We expect information about disks to be filled and will try to adjust disks states accordingly
+//If we want change state to online/offline we should define its state
+//If we want to add disk we should define it here and set DiskConfigType to online or offline
+//If we want to remove disk we should set its state to removed
```

Comment:
Did you mean "unused" instead of "removed"?

Reply:
modified